common/network-common/src/main/java/org/apache/spark/network/TransportContext.java

```diff
@@ -187,7 +190,11 @@ public TransportChannelHandler initializePipeline(
       RpcHandler channelRpcHandler) {
     try {
       TransportChannelHandler channelHandler = createChannelHandler(channel, channelRpcHandler);
-      ChannelPipeline pipeline = channel.pipeline()
+      ChannelPipeline pipeline = channel.pipeline();
+      if (nettyLogger.getLoggingHandler() != null) {
+        pipeline.addLast("loggingHandler", nettyLogger.getLoggingHandler());
```

Comment:
Just a question, this is intentionally added at the first, right?

Reply:
Yes as in this case we are logging what goes into the wire and not what goes between two channel handers.